python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -533,6 +537,12 @@ def has_seen_output(self, output_name: str, mapping_key: Optional[str] = None) -
             )
         return output_name in self._seen_outputs
 
+    def asset_partition_key_for_output(self, output_name: str = "result") -> str:
```

Comment:
I think it might be better to override the `partition_key` property rather than this function

Reply:
@OwenKephart  I attempted to do that and got an error 

```
 AttributeError: 'BoundOpExecutionContext' object has no attribute '_step_execution_context'. Did you mean: 'get_step_execution_context'?
```

Is the BoundOpExecutionContext supposed to have a step execution context?

Reply:
Hm I think this is a bit tricky, as I would expect users to more often call `context.partition_key` in order to get that information in the context of a run. But it looks like we support a few different ways for users to branch behavior inside the body of the asset based off of partition information.

Maybe both need to be overridden? @sryza

Reply:
I added an update that implements the property and method in both the unbound and bound context plus test coverage.